segmentation/include/pcl/segmentation/impl/approximate_progressive_morphological_filter.hpp

```diff
@@ -129,7 +130,9 @@ pcl::ApproximateProgressiveMorphologicalFilter<PointT>::extract (Indices& ground
   for (int i = 0; i < static_cast<int>(input_->size ()); ++i)
   {
     // ...then test for lower points within the cell
-    PointT p = (*input_)[i];
+    const PointT& p = (*input_)[i];
+    if (!pcl::isFinite(p))
```

Comment:
Don't we need to test at line 235 as well - ground indices is the orignally indices input (so could be all points, if not thing is set), which results in a copy of the original cloud, iterated at the mentioned line?

Reply:
Should we only keep these checks for algorithms that require ordered point clouds and leave them out of algorithms, that works on unordered - to avoid the performance penalty of checking all points (and this case, twice?)

Reply:
However, there is no way to inform the user that its a ill-formed cloud, if we don't check them... so I guess it would bring these issues once in a while.
I guess the best option is to add a "is dense" version of all algorithms, where the user knows what is used as input.

Reply:
Like this?

Reply:
Yeah, exactly :smile: